services/web/server/src/simcore_service_webserver/folders/_folders_api.py

```diff
@@ -1,67 +1,91 @@
 # pylint: disable=unused-argument
 
 import logging
-from typing import cast
 
 from aiohttp import web
-from aiopg.sa.engine import Engine
-from models_library.api_schemas_webserver.folders import FolderGet, FolderGetPage
+from models_library.api_schemas_webserver.folders_v2 import FolderGet, FolderGetPage
 from models_library.folders import FolderID
 from models_library.products import ProductName
-from models_library.projects_access import AccessRights
 from models_library.rest_ordering import OrderBy
-from models_library.users import GroupID, UserID
-from pydantic import NonNegativeInt, parse_obj_as
-from simcore_postgres_database import utils_folders as folders_db
+from models_library.users import UserID
+from models_library.workspaces import WorkspaceID
+from pydantic import NonNegativeInt
+from simcore_service_webserver.workspaces._workspaces_api import (
+    check_user_workspace_access,
+)
 
-from .._constants import APP_DB_ENGINE_KEY
 from ..users.api import get_user
+from ..workspaces.errors import (
+    WorkspaceAccessForbiddenError,
+    WorkspaceFolderInconsistencyError,
+)
+from . import _folders_db as folders_db
 
 _logger = logging.getLogger(__name__)
 
 
 async def create_folder(
     app: web.Application,
     user_id: UserID,
-    folder_name: str,
-    description: str | None,
+    name: str,
     parent_folder_id: FolderID | None,
     product_name: ProductName,
+    workspace_id: WorkspaceID | None,
```

Comment:
so for me the name change is not helpful, I can see it from the typing already.
you use a lot of differnet terms in here, I think it would help to have a bit of consistency. For example:
- private vs personal


If I get it right, if the function is called with a workspace ID, you first check if this is a shared or private workspace?
What about creating a function such as, would that not be easier to read?:
```python
if workspace_id:
  check_workspace_access(required_access: read | write | delete)
  workspace_is_private = True
  ...
```

Since you are re-using this pattern a lot, I guess that would reduce the size of each function by a bunch no?